src/content/docs/en/guides/cms/builderio.mdx

```diff
@@ -3,19 +3,369 @@ title: Builder.io & Astro
+const builderAPIpublicKey = import.meta.env.BUILDER_API_PUBLIC_KEY;
+// Builder's API requires this field but for this use case the url doesn't seem to matter - the API returns the same HTML
+const encodedUrl = encodeURIComponent("moot");
+const { html: postHTML } = await fetch(
+  `https://cdn.builder.io/api/v1/qwik/${builderModel}?${new URLSearchParams({
+    apiKey: builderAPIpublicKey,
+    url: encodedUrl,
+    "query.data.slug": slug,
+    cachebust: "true",
+  }).toString()}`
+)
+  .then((res) => res.json())
+  .catch();
+---
+<html lang="en">
+  <head>
+    <title>{title}</title>
+  </head>
+  <body>
+    <header>This is your header</header>
+    <article>
+      <Fragment set:html={postHTML} />
+    </article>
+    <footer>The is your footer</footer>
+  </body>
+</html>
+```
+
+:::note
+The variables `builderModel` and `builderAPIpublicKey` need to be created twice, since [`getStaticPaths()` runs in its own isolated scope](https://docs.astro.build/en/reference/api-reference/#getstaticpaths).
+:::
+
+Now when you click on a link on your index route, you will be taken to the individual blog post page.
+
+### Publishing your site
+
+To deploy your website, visit our [deployment guides](/en/guides/deploy/) and follow the instructions for your preferred hosting provider.
+
+#### Rebuild on Builder changes
+
+If your project is using Astro's default static mode, you will need to set up a webhook to trigger a new build when your content changes. If you are using Netlify or Vercel as your hosting provider, you can use its webhook feature to trigger a new build whenever you click **Publish** in the Builder editor.
+
+##### Netlify
+
+1. Go to your site dashboard, then **Site Settings** and click on **Build & deploy**.
+2. Under the **Continuous Deployment** tab, find the **Build hooks** section and click on **Add build hook**.
+3. Provide a name for your webhook and select the branch you want to trigger the build on. Click on **Save** and copy the generated URL.
+
+##### Vercel
+
+1. Go to your project dashboard and click on **Settings**.
+2. Under the **Git** tab, find the **Deploy Hooks** section.
+3. Provide a name for your webhook and the branch you want to trigger the build on. Click **Add** and copy the generated URL.
+
+##### Adding a webhook to Builder
+
+:::tip[Official resource]
+Find Builder's guide on adding webhooks [here](https://www.builder.io/c/docs/webhooks).
+:::
+
+1. In your Builder dashboard, go into your **`blogpost`** model. Under **Show More Options**, select **Edit Webhooks** at the bottom.
+2. Add a new webhook by clicking on **Webhook**. Paste the URL generated by your hosting provider into the **Url** field.
+3. Click on **Show Advanced** under the URL field and toggle the option to select **Disable Payload**. With the payload disabled, Builder sends a simple POST request to your hosting provider. Otherwise, the payload might overwhelm your hosting provider, as your site grows. Click **Done** to save this selection.
```

Comment:
> Otherwise, the payload might overwhelm your hosting provider

OK, I can see this is a change I made, that you reverted and added back. I'm not gonna lie, I really dislike this. :smile:

The first time I read this, I thought, "If I need to toggle this option on, because otherwise Builder is going to bork my hosting provider, a) why isn't it on by default and b) uh... why did Builder build something that might break my hosting?"

So, it's not just about being explanatory. It's also about, "Is something going to *distract the reader* as they're trying to complete the steps?" This absolutely distracted me, and worse, gave me a bad impression *of Builder*. (There may be very good reasons behind all this, but again, that would be more that you'd need to put here that are not relevant to just following the instructions and getting it working.) That's why I had a more neutral statement "do this" here.

But, it's your show! I just want to make you aware of that!